lib/src/core/repositories/vocabulary_repository.dart

```diff
@@ -14,4 +14,23 @@ class VocabularyRepository {
 
     return vocabulary;
   }
+
+  Future<List<Vocabulary>> searchVocabularyRomaji(String searchTxt) async {
+    return _vocabulary
+        .where((vocabulary) =>
+            vocabulary.romaji.contains(searchTxt) ||
+            vocabulary.meanings
+                .where((meaning) => meaning.contains(searchTxt))
```

Comment:
Ici tu pourrais utiliser [indexWhere](https://api.dart.dev/stable/3.2.3/dart-core/List/indexWhere.html) plutôt qui serait moins gourmand ou [lastIndexWhere](https://api.dart.dev/stable/3.2.3/dart-core/List/lastIndexWhere.html)

```suggestion
                .lastIndexWhere((meaning) => meaning.contains(searchTxt)) >= 0
```

Reply:
Changé